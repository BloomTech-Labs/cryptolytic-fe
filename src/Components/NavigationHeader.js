import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import cogWheel from "../images/cogwheel.svg";
import Avatar from "@material-ui/core/Avatar";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import firebase from "../firebase";

const useStyles = makeStyles(theme => ({
	navHeader: {
		backgroundColor: "#000000",
		width: "100%",
		height: "10vh",
		display: "flex",
		justifyContent: "space-between",
		"@media(max-width: 1300px)": {
			display: "flex",
			justifyContent: "space-around",
			width: "15%"
		}
	},
	title: {
		width: "20vw",
		height: "10vh",
		marginLeft: "18vw",
		display: "flex",
		justifyContent: "center",
		alignItems: "center",
		color: "#FFFFFF",
		fontWeight: 300,
		fontSize: "30px",
		lineHeight: "46px",
		letterSpacing: "0.03em",
		marginBottom: "30vw",
		"@media(max-width: 1300px)": {
			display: "none"
		}
	},
	cogWheel: {
		width: "10%",
		height: "10vh",
		display: "flex",
		justifyContent: "space-evenly",
		alignItems: "center",
		"@media(max-width: 1300px)": {
			width: "100%"
		}
	}
}));

function NavigationHeader({ history }) {
	const [anchorEl, setAnchorEl] = useState(null);
	const classes = useStyles();

	const handleClick = event => {
		setAnchorEl(event.currentTarget);
	};

	const handleClose = () => {
		setAnchorEl(null);
	};

	const handleLogout = () => {
		localStorage.removeItem("firebase_jwt");
		firebase.auth().signOut();
		history.push("/");
	};

	return (
		<div className={classes.navHeader}>
			<div className={classes.title}></div>
			<div className={classes.cogWheel}>
				<img src={cogWheel} />
				<div>
					<Avatar
						aria-controls='profile-menu'
						aria-haspopup='true'
						onClick={handleClick}
						style={{ cursor: "pointer" }}
					/>
					<Menu
						id='profile-menu'
						anchorEl={anchorEl}
						keepMounted
						open={Boolean(anchorEl)}
						onClose={handleClose}
					>
						<MenuItem onClick={handleLogout}>Logout</MenuItem>
					</Menu>
				</div>
			</div>
		</div>
	);
}

export default NavigationHeader;
