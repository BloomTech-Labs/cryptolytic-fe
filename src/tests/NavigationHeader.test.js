import React from "react";
///
import NavigationHeader from "../Components/NavigationHeader";
///
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

test("NavigationHeader component initialized without crashing", () => {
  render(<NavigationHeader />);
});

describe("<NavigationHeader /> matches verified SnapShot", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<NavigationHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
