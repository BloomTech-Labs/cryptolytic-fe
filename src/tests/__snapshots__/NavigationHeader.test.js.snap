// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationHeader /> matches verified SnapShot should match snapshot 1`] = `
<div
  className="makeStyles-navHeader-20"
>
  <div
    className="makeStyles-title-21"
  />
  <div
    className="makeStyles-cogWheel-22"
  >
    <img
      src="cogwheel.svg"
    />
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
</div>
`;
